src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -357,15 +354,18 @@ private async Task UpdateAdvertisingManifestAsync(ManifestId manifestId, bool in
                 {
                     File.Delete(packagePath);
                 }
-
-                var versionDir = Path.GetDirectoryName(packagePath);
-                if (Directory.Exists(versionDir) && !Directory.GetFileSystemEntries(versionDir).Any())
+                if (!string.IsNullOrEmpty(packagePath))
```

Comment:
I think we only want to do this Directory cleanup if we were *not* using the offline cache.

Reply:
@dsplaisted so should this not check anything regarding packagePath and instead only check if we are using the offline cache? Or should the offline cache check be added?